store/mongo/datastore_mongo.go

```diff
@@ -1530,6 +1530,20 @@ func (db *DataStoreMongo) GetDevicesListForDeployment(ctx context.Context,
 					Value: model.DeviceDeploymentStatusPending,
 				}},
 			})
+		} else if *q.Status == model.DeviceDeploymentStatusFinishedStr {
```

Comment:
we have `active` and `finished` fields in the device deployment object, maybe we could use them to create a query for active and finished device deployments?

Reply:
I believe the reason why we query by status is because it uses the index in MongoDB.
Querying by other fields requires additional indexes.

Reply:
you're right; we don't have indexes for `active` and `finished`;
I thought we introduced `active` to improve some queries and we introduced index with that field, but, after checking, looks like it's not the case;
sorry for the noise;